sim-administration/sim-batch-management/es2plus/es2plus.go

```diff
@@ -0,0 +1,401 @@
+	// Return an error if the bytes can't be retrieved.
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	responseBytes := []byte(response)
+	return responseBytes, nil
+}
+
+
+///
+///  Externally visible API for Es2Plus protocol
+///
+
+
+func GetStatus(client *Es2PlusClient, iccid string) (*ProfileStatus, error) {
+    result := new(ES2ProfileStatusResponse)
+    es2plusCommand := "getProfileStatus"
+    header := newEs2plusHeader(client)
+    payload := &ES2PlusGetProfileStatusRequest{
+               		Header:    *header,
+               		IccidList: [] ES2PlusIccid{ES2PlusIccid{Iccid: iccid}},
+               	}
+    err := marshalUnmarshalGenericEs2plusCommand(client, es2plusCommand, payload, result)
+    if err != nil {
+        return nil, err
+    }
+
+    if (len(result.ProfileStatusList) == 0) {
+        return nil, nil
+    } else if (len(result.ProfileStatusList)  == 1) {
+        returnvalue := result.ProfileStatusList[0]
+        return &returnvalue, nil
+    } else {
+       return nil, errors.New("GetStatus returned more than one profile!")
+    }
+}
+
+
+func RecoverProfile(client *Es2PlusClient, iccid string, targetState string) (*ES2PlusRecoverProfileResponse, error) {
+    result := new(ES2PlusRecoverProfileResponse)
+    es2plusCommand := "recoverProfile"
+    header := newEs2plusHeader(client)
+    payload := &ES2PlusRecoverProfileRequest{
+               		Header:        *header,
+               		Iccid:         iccid,
+               		ProfileStatus: targetState,
+               	}
+    return result, marshalUnmarshalGenericEs2plusCommand(client, es2plusCommand, payload, result)
+}
+
+
+func CancelOrder(client *Es2PlusClient, iccid string, targetState string) (*ES2PlusCancelOrderResponse, error) {
+    result := new(ES2PlusCancelOrderResponse)
+    es2plusCommand := "cancelOrder"
+    header := newEs2plusHeader(client)
+    payload := &ES2PlusCancelOrderRequest {
+               		Header:        *header,
+               		Iccid:         iccid,
+               		FinalProfileStatusIndicator: targetState,
+               	}
+    return result, marshalUnmarshalGenericEs2plusCommand(client, es2plusCommand, payload, result)
+}
+
+func DownloadOrder(client *Es2PlusClient, iccid string) (*ES2PlusDownloadOrderResponse, error) {
+    result := new(ES2PlusDownloadOrderResponse)
+    es2plusCommand := "downloadOrder"
+    header := newEs2plusHeader(client)
+    payload := &ES2PlusDownloadOrderRequest {
+               		Header:        *header,
+               		Iccid:         iccid,
+               		Eid:           "",
+               		Profiletype:   "",
+               	}
+    err :=  marshalUnmarshalGenericEs2plusCommand(client, es2plusCommand, payload, result)
+    if err != nil {
+            return nil, err
+    }
+
+    executionStatus := result.Header.FunctionExecutionStatus.FunctionExecutionStatusType
+    if ("Executed-Success" != executionStatus) {
```

Comment:
There's no need for these unidiomatic backwards comparisons because assignment in Go is not an expression, only a statement.